fs/fs_zenfs.cc

```diff
@@ -516,16 +516,23 @@ IOStatus ZenFS::FileExists(const std::string& fname, const IOOptions& options,
   }
 }
 
+/* If the file does not exist, create a new one,
+ * else return the existing file
+ */
 IOStatus ZenFS::ReopenWritableFile(const std::string& fname,
                                    const FileOptions& options,
                                    std::unique_ptr<FSWritableFile>* result,
                                    IODebugContext* dbg) {
+  std::shared_ptr<ZoneFile> zoneFile = GetFile(fname);
+
   Debug(logger_, "Reopen writable file: %s \n", fname.c_str());
 
-  if (GetFile(fname) != nullptr)
+  if (zoneFile == nullptr)
     return NewWritableFile(fname, options, result, dbg);
-
-  return target()->NewWritableFile(fname, options, result, dbg);
+  else
+    result->reset(new ZonedWritableFile(zbd_, !options.use_direct_writes,
+                                        zoneFile, &metadata_writer_));
```

Comment:
We have some overlap of code in `NewWritableFile()`, `ReopenWritableFile()`. Could we not make both delegate to a protected `OpenFile()`? I think that is what the posix layer is doing, we could model after that.

Reply:
Thanks, good suggestion. Done.